### PR TITLE
Cursor trails

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Game/Graphics/Cursor/CursorTrail.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        {
+            Origin = Anchor.Centre;
+        }
+
+        protected override void ApplyDrawNode(DrawNode node)
+        {
+            base.ApplyDrawNode(node);
+
+            SpriteDrawNode sNode = node as SpriteDrawNode;
+            sNode.RoundedTextureShader = sNode.TextureShader = shader;
```
Comment (UselessToucan) [inline osu.Game/Graphics/Cursor/CursorTrail.cs]: Using `as` without null check(or null propagation) is a bit weird. I think it should be just `(SpriteDrawNode) node`.

File: osu.Game/Graphics/Cursor/CursorTrail.cs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+        Vector2? lastPosition;
+
+        protected override DrawNode CreateDrawNode() => new TrailDrawNode();
+
+        protected override void ApplyDrawNode(DrawNode node)
+        {
+            base.ApplyDrawNode(node);
+
+            TrailDrawNode tNode = node as TrailDrawNode;
```
Comment (UselessToucan) [inline osu.Game/Graphics/Cursor/CursorTrail.cs]: `as` without handling null >_<

File: osu.Game/Graphics/Cursor/CursorTrail.cs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            shader = shaders?.Load(@"CursorTrail", FragmentShaderDescriptor.Texture);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            Invalidate(Invalidation.DrawNode);
+
+            int fadeClockResetThreshold = 1000000;
```
Comment (UselessToucan) [inline osu.Game/Graphics/Cursor/CursorTrail.cs]: Magic number?

File: osu.Game/Graphics/Cursor/CursorTrail.cs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+            time = (float)(Time.Current - timeOffset) / 500f;
```
Comment (UselessToucan) [inline osu.Game/Graphics/Cursor/CursorTrail.cs]: Magic number?

File: osu.Game/Graphics/Cursor/CursorTrail.cs (hunk 3 above)
Comment (peppy) [inline osu.Game/Graphics/Cursor/CursorTrail.cs]: yes

File: osu.Game/Graphics/Cursor/CursorTrail.cs (hunk 4 above)
Comment (peppy) [inline osu.Game/Graphics/Cursor/CursorTrail.cs]: yes

**Pull request merged**